### PR TITLE
feat(sessions): add Command Code as a tracked source

### PR DESCRIPTION
**Pull request by baanish**
## What
Adds Command Code (commandcode.ai) as a tracked session source, scanning `~/.commandcode/projects/<slug>/*.jsonl`.

## Why it's estimated
Command Code does **not** persist token usage on disk — the CLI computes usage in memory and ships it to its backend (confirmed by decompiling the npm bundle; the local transcript carries only message text). So tokens are estimated from the transcript, the same way tokscale already handles `kiro`:
- **input** = cumulative conversation context preceding each assistant turn (Command Code re-sends the full context per request)
- **output** = that turn's own content
- ~4 chars/token, counted from each message's canonical JSON serialization so tool args/results are included

The model id comes from `~/.commandcode/config.json` and is canonicalized (e.g. `MiniMaxAI/MiniMax-M3-Free` → `MiniMax-M3`) so pricing resolves to the real paid model instead of the free-promo entry or a fuzzy mismatch. The source bypasses the parse cache because that model dependency lives outside the fingerprinted transcript file.

## Known limitation
The estimate approximates tokens *processed* and will **not** match Command Code's server dashboard: it counts full tool outputs that the tool truncates before billing, and cannot see auxiliary (tool-desc / taste-1) runs. On a real ~880-turn session it estimates **270M input / ~\$81** at the MiniMax-M3 rate, versus the dashboard's 181M / \$0.00 (a promo). This is consistent with how tokscale treats its other estimated sources.

## Testing
- 9 parser unit tests (estimation, model canonicalization, checkpoint/malformed/empty edge cases)
- registry + CLI mirror-point tests updated (`ClientId::COUNT`, `ClientFilter`, TUI client list)
- real-session smoke via `tokscale clients` and `tokscale models -c commandcode`
- full suites green: core 927, CLI 778

Reviewed adversarially with Codex (2 rounds); fixes for model-pricing resolution and config-aware (cache-bypass) parsing folded in.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Related to #668 

---
## Summary by cubic
Add `commandcode` as a tracked session source with estimated token usage and pricing so Command Code sessions show up with costs and analytics. Scans local transcripts and applies model-aware pricing without relying on server usage; registered as client id 29.

- **New Features**
  - Scan `~/.commandcode/projects/<slug>/*.jsonl`; skip `*.checkpoints.jsonl`.
  - Estimate tokens from canonical JSON (~4 chars/token): input = cumulative context before each assistant turn; output = that turn’s content; `cache_read = 0`.
  - Read model from `~/.commandcode/config.json`, canonicalize for pricing (e.g., `MiniMaxAI/MiniMax-M3-Free` → `MiniMax-M3`), and bypass the parse cache so config changes take effect immediately.
  - Register across the app: CLI filter flag `--commandcode` and name mapping, TUI entry “Command Code” with hotkey `d`, client id 29 (`ClientId::COUNT` = 30), and frontend display name/logo/color.

<sup>Written for commit 9c1bcd1d0af6fcb537adfa3256c89fb07e391018. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/junhoyeo/tokscale/pull/717?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: crates/tokscale-core/src/sessions/commandcode.rs [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+
+        if session_id.is_none() {
+            if let Some(id) = entry.session_id.as_deref().filter(|id| !id.is_empty()) {
+                session_id = Some(id.to_string());
+            }
+        }
+
+        let chars = entry.content.as_ref().map(content_chars).unwrap_or(0);
+
+        match entry.role.as_deref() {
```
[the remark this answers is not in the document]
Comment (baanish) [reply crates/tokscale-core/src/sessions/commandcode.rs:121]: > _Note: this reply is posted by Claude Code on behalf of the author, not written by them._

This finding doesn't apply to Command Code's format. Command Code transcripts use a top-level `role` field (`user` / `assistant` / `tool`), **not** a top-level `type`. The `type` field only appears *inside* content parts (e.g. `{"type":"text", ...}`), which the `match entry.role` here never touches.

Verified against a real session (1,766 lines): every line carries `role` (880 `assistant`, 862 `tool`, 24 `user`), and **zero** lines have a top-level `type`. Running it end to end, `tokscale models -c commandcode` reports ~270M input across 880 messages — not zero usage.

The suggestion looks like it assumed the OpenAI/Codex transcript shape, which does use a top-level `type`.
